src/lib/gadgets/gadgets.ts

```diff
@@ -106,6 +106,72 @@ const Gadgets = {
   xor(a: Field, b: Field, length: number) {
     return xor(a, b, length);
   },
+
+  /**
+   * Performs a left shift operation on the provided {@link Field} element.
+   * This operation is similar to the `<<` shift operation in JavaScript,
+   * where bits are shifted to the left, and the overflowing bits are discarded.
+   *
+   * It’s important to note that these operations are performed considering the big-endian 64-bit representation of the number,
+   * where the most significant (64th) bit is on the left end and the least significant bit is on the right end.
+   *
+   * **Important:** The gadgets assumes that its input is at most 64 bits in size.
+   *
+   * If the input exceeds 64 bits, the gadget is invalid and does not prove correct execution of the shift.
```

Comment:
If the input exceeds 64 bit, does proving fail or does the gadget provide invalid results?

Reply:
I think the former. I just tried it as a sanity check and I get a prover error:
`Error: the proof could not be constructed: rest of division by vanishing polynomial`

Reply:
But that's because you're not maliciously changing the witness!! If you'd modify the witness blocks, and could also control the inputs (and the circuit doesn't prevent them from exceeding 64 bits) you could create a proof and get the gadget to return invalid results

Reply:
Anyway, the updated wording also reads good to me 👍🏻